mesonbuild/cargo/interpreter.py

```diff
@@ -151,7 +150,43 @@ class Package:
     autoexamples: bool = True
     autotests: bool = True
     autobenches: bool = True
+    api: str = dataclasses.field(init=False)
 
+    def __post_init__(self) -> None:
+        self.api = _version_to_api(self.version)
+
+@dataclasses.dataclass
+class SystemDependency:
+
+    """ Representation of a Cargo system-deps entry
+        https://docs.rs/system-deps/latest/system_deps
+    """
+
+    name: str
+    version: T.List[str]
+    optional: bool
+    feature: T.Optional[str]
+    feature_overrides: T.Dict[str, T.Dict[str, str]]
+
+    @classmethod
+    def from_raw(cls, name: str, raw: T.Any) -> SystemDependency:
```

Comment:
```suggestion
    def from_raw(cls, name: str, raw: T.Dict[str, T.Union[str, T.Dict[str, T..Any]]) -> SystemDependency:
```
Would still be an improvement, though it would be nicer if we actually annotated this, which would help us with our expectations. But looking at the documentation it's fairly non-trival.

Which, actually, it doesn't look like this handles the case of:
```toml
[package.metadata.system-deps]
testlib = "1.2"
```

Reply:
I think `raw: T.Dict[str, T.Any]` is the most you can really tell.

Reply:
Fixed the `testlib = "1.2"` case.